rewrite-java/src/main/java/org/openrewrite/java/cleanup/SimplifyConstantIfBranchExecution.java

```diff
@@ -47,75 +48,94 @@ public String getDescription() {
 
     private static class SimplifyConstantIfBranchExecutionVisitor extends JavaVisitor<ExecutionContext> {
 
-        @Override
-        public J visitWhileLoop(J.WhileLoop whileLoop, ExecutionContext executionContext) {
-            return whileLoop;
-        }
-
         @Override
         public J visitBlock(J.Block block, ExecutionContext executionContext) {
             J.Block bl = (J.Block) super.visitBlock(block, executionContext);
-            List<Statement> addStatements = getCursor().pollMessage("statements");
-            J.If removeIf = getCursor().pollMessage("remove-if");
-            if (removeIf != null) {
-                bl = maybeAutoFormat(bl, bl.withStatements(ListUtils.flatMap(bl.getStatements(), stmt -> {
-                    if (stmt == removeIf) {
-                        return addStatements;
-                    }
-                    return stmt;
-                })), executionContext, getCursor().getParent());
+            if (bl != block) {
+                bl = (J.Block) new RemoveUnneededBlock.RemoveUnneededBlockStatementVisitor()
+                    .visitNonNull(bl, executionContext, getCursor().getParentOrThrow());
+                EmptyBlockStyle style = ((SourceFile) getCursor().firstEnclosingOrThrow(JavaSourceFile.class))
```

Comment:
`visitBlock` is a fairly hot path, consider calculating the style only once by adding a local variable and getter that will only calculate the style if its null.

Reply:
in hindsight, the EmtpyBlockStyle is only created when a change is detected so this actually not that expensive

Reply:
in hindsight, the EmtpyBlockStyle is only created when a change is detected so this actually not that expensive